client/blockchain-service/src/handler.rs

```diff
@@ -1046,4 +1092,49 @@ impl BlockchainService {
         };
         current_tick == &last_tick_provided.saturating_add(provider_challenge_period)
     }
+
+    fn check_pending_forest_root_writes(&mut self) {
+        if let Some(mut rx) = self.forest_root_write_lock.take() {
+            // TODO: consider adding a timeout here to prevent a task from blocking the runtime.
```

Comment:
What do you mean by "runtime" here? The node running? I'm asking because "runtime" in a substrate node is a loaded word.